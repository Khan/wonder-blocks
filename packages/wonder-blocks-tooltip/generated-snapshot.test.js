// @flow
// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests
// edit packages/wonder-blocks-tooltip/docs.md and run `npm run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

describe("wonder-blocks-tooltip", () => {
    it("example 1", () => {
        const React = require("react");
        const {View} = require("@khanacademy/wonder-blocks-core");
        const {StyleSheet} = require("aphrodite");
        const {Tooltip} = require("@khanacademy/wonder-blocks-tooltip");
        const {
            StandardModal,
            ModalLauncher,
        } = require("@khanacademy/wonder-blocks-modal");

        const tooltip = (
            <Tooltip content={"Some long text"}>
                <View>Anchor</View>
            </Tooltip>
        );

        const scrollyContent = (
            <div
                style={{
                    height: 100,
                    overflow: "auto",
                    border: "1px solid",
                    margin: 10,
                }}
            >
                <div style={{height: "200vh"}}>{tooltip}</div>
            </div>
        );

        const modalContent = (
            <div style={{height: "200vh"}}>{scrollyContent}</div>
        );

        const modal = (
            <StandardModal
                title="My modal"
                footer="Still my modal"
                content={modalContent}
            />
        );

        const example = (
            <div>
                {tooltip}

                {scrollyContent}

                <ModalLauncher modal={modal}>
                    {({openModal}) => (
                        <button onClick={openModal}>Click here!</button>
                    )}
                </ModalLauncher>
            </div>
        );

        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
