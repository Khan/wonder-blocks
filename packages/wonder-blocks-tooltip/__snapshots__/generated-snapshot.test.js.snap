// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-tooltip example 1 1`] = `
<div>
  <div />
  <div
    style={
      Object {
        "border": "1px solid",
        "height": 100,
        "margin": 10,
        "overflow": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "height": "200vh",
        }
      }
    >
      <div />
    </div>
  </div>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <button
      onClick={[Function]}
    >
      Click here!
    </button>
  </div>
</div>
`;
